### ISSUE TITLE
Geopackage-provided tiles are malformed in v20

### ISSUE THREAD

**Issue by Monstrofil** [go-spatial/tegola#999]
I have this minimal config:
```
tile_buffer = 64

[webserver]
port = ":9090"

[[providers]]
name = "sample_gpkg"
type = "gpkg"
filepath = "bp.gpgk"
srid = 4326

[[providers.layers]]
name = "landuse1"
tablename = "bp"
geometry_fieldname = "geometry"
id_fieldname = "fid"

[[maps]]
name = "landuse_new"

[[maps.layers]]
provider_layer = "sample_gpkg.landuse1"
dont_simplify = true
min_zoom = 3
max_zoom = 15

```

When running with gospatial/tegola:v0.20.0, I get this:
![image](https://github.com/go-spatial/tegola/assets/174041930/a813e235-0f44-4804-b66f-ae2883ce7362)

When i switch to gospatial/tegola:v0.19.0, geometry becomes normal:
![image](https://github.com/go-spatial/tegola/assets/174041930/87a6d2c1-41fb-4719-9a14-e9871213332a)

I believe there is a bug in v20.

**Comment by iwpnd** [comment]
Can you share the file please?

**Comment by Monstrofil** [comment]
Sure.
[demo.zip](https://github.com/user-attachments/files/16088523/demo.zip)

**Comment by iwpnd** [comment]
I can confirm the issue in v0.20.0 and everything working in v0.19.0. Also took a look at the data and couldn't find anything wrong with it either. I was able to verify that the gpkg provider correctly gets the feature from the gpkg and creates a valid polygon. I'm not sure why, but the issues appears to occur during encoding to mvt.

**Comment by ARolek** [comment]
I have a feeling this has to do with this PR https://github.com/go-spatial/tegola/pull/952. 

@gdey can you give us some guidance on what our best path forward here is? It seems to me that upgrading the geom package has caused a regression in rendering. I think the geom package version was bumped to bring in a fix for the slippy package to get tests to pass on Apple Silicon. This was discussed in https://github.com/go-spatial/tegola/issues/951

**Comment by gdey** [comment]
I need to look into this, and the other one listed here. I'll take a look tomorrow. See what I can find, but it does look like an issue with geom.

**Comment by iwpnd** [comment]
Just for a sanity check. The mvt_postgis provider is not affected.

**Comment by ARolek** [comment]
> Just for a sanity check. The mvt_postgis provider is not affected.

Correct. The `mvt_postgis` provider does all geoprocessing in PostGIS. The Geopackage provider does the geoprocessing in tegola, which is why this is showing up only in the geopackage provider.

**Comment by iwpnd** [comment]
> which is why this is showing up only in the geopackage provider.

just tested the legacy  `postgis` provider. Same issue there for `geometry_type=polygon` but fine for `geometry_type=linestring`. 👀

**Comment by ARolek** [comment]
@iwpnd indeed, the same issue would show up for the legacy postgis provider. The geom package's line handling is fine, but the polygon handling is where the issues stem from.

**Comment by gdey** [comment]
Yeah, this looks like an old regression I had with makevalid. Where it would drop certain points due to floating point precision errors.

**Comment by ARolek** [comment]
Note, @gdey has been working on a fix in the geom package and hopes to have a PR ready soon that should address this issue.

**Comment by gdey** [comment]
I'm still looking at this. Need to take a small break, but am look at it.

**Comment by tommyannguyen** [comment]
confirm v20 data has broken . I revert v19 work ok

**Comment by gdey** [comment]
Sorry, this is taking a bit. Had to stop to get a few other things done, this change is harder to do that I was expecting. Upgrading geom to v1.0.0 and caused this be a major change. Need to make the changes to geom, so overall I think it's good. Just this is taking a bit.

**Comment by fabermichiel** [comment]
This happens for me with the postgis provider. Converting to mvt_postgis resolves the problem for my use case
